src/relayer/Relayer.ts

```diff
@@ -36,13 +36,53 @@ export class Relayer {
     // TODO: Note this does not consider the price of the token which will be added once the profitability module is
     // added to this bot.
 
+    // Sum the total unfilled deposit amount per origin chain and set a MDC for that chain.
+    const unfilledDepositAmountsPerChain: { [chainId: number]: BigNumber } = getUnfilledDeposits(
+      this.clients.spokePoolClients,
+      this.maxUnfilledDepositLookBack
+    ).reduce((agg, curr) => {
+      const unfilledAmountUsd = this.clients.profitClient.getFillAmountInUsd(curr.deposit, curr.unfilledAmount);
+      if (!agg[curr.deposit.originChainId]) agg[curr.deposit.originChainId] = toBN(0);
+      agg[curr.deposit.originChainId] = agg[curr.deposit.originChainId].add(unfilledAmountUsd);
+      return agg;
+    }, {});
+
+    // Sort thresholds in ascending order.
+    const minimumDepositConfirmationThresholds = Object.keys(this.config.minDepositConfirmations)
+      .filter((x) => x !== "default")
+      .sort((x, y) => Number(x) - Number(y));
+
+    // Set the MDC for each origin chain equal to lowest threshold greater than the unfilled USD deposit amount.
+    // If we can't find a threshold greater than the USD amount, then use the default.
+    const mdcPerChain = Object.fromEntries(
+      Object.keys(unfilledDepositAmountsPerChain).map((chainId) => {
+        const usdThreshold = minimumDepositConfirmationThresholds.find((_usdThreshold) => {
+          return toBNWei(_usdThreshold).gte(unfilledDepositAmountsPerChain[chainId]);
```

Comment:
Nit: up to you, but I find using entries to be simpler than re-accessing the struct. I think it's also a bit easier to guarantee that the assessed value actually exists (i.e. isn't undefined).

```suggestion
    const mdcPerChain = Object.fromEntries(
      Object.entries(unfilledDepositAmountsPerChain).map(([chainId, unfilledAmount]) => {
        const usdThreshold = minimumDepositConfirmationThresholds.find((_usdThreshold) => {
          return toBNWei(_usdThreshold).gte(unfilledAmount);
```

Reply:
Yeah its cleaner good call out